Kitodo/src/main/java/de/sub/goobi/forms/ProjekteForm.java

```diff
@@ -800,4 +843,118 @@ public int getItemId() {
         return this.itemId;
     }
 
+    /**
+     * Get the XML configuration string.
+     *
+     * @return the XML configuration string
+     */
+    public String getXMLConfiguration() {
+        return this.xmlConfigurationString;
+    }
+
+    /**
+     * Set the XML configuration string.
+     *
+     * @param configuration
+     *            the XML configuration string
+     */
+    public void setXMLConfiguration(String configuration) {
+        this.xmlConfigurationString = configuration;
+    }
+
+    /**
+     * Load the content of the XML configuration file denoted by 'configurationFile'
+     * as a String.
+     */
+    public void loadXMLConfiguration(String configurationName) {
+        try {
+            currentConfigurationFile = configurationFiles.get(configurationName);
+            XMLConfiguration currentConfiguration = new XMLConfiguration(
+                    ConfigCore.getKitodoConfigDirectory() + currentConfigurationFile);
+            StringWriter stringWriter = new StringWriter();
+            currentConfiguration.save(stringWriter);
+            this.xmlConfigurationString = stringWriter.toString();
+        } catch (ConfigurationException e) {
+            String errorMessage = "ERROR: Unable to load configuration file for '" + configurationName + "'.";
+            logger.error(errorMessage + " " + e.getMessage());
+            this.xmlConfigurationString = errorMessage;
+        }
+    }
+
+    /**
+     * Save the String 'xmlConfigurationString' to the XML file denoted by
+     * 'configurationFile'.
+     */
+    public void saveXMLConfiguration() {
+        try {
+            Document document = documentBuilder.parse(new InputSource(new StringReader(this.xmlConfigurationString)));
+            TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
+            DOMSource domSource = new DOMSource(document);
+            File xmlConfigurationFile = new File(ConfigCore.getKitodoConfigDirectory() + currentConfigurationFile);
+            StreamResult streamResult = new StreamResult(
+                    new PrintWriter(new FileOutputStream(xmlConfigurationFile, false)));
```

Comment:
Opening a lot of stream object without closing it? This looks like a possible resource leak. Could this streams closed after usage? If the answer is yes, then should it possible to use try-with-resources statement to write less code.